src/packet/mod.rs

```diff
@@ -10,6 +10,22 @@
 
 #![macro_use]
 
+/// Used to help compute checksum value of packet
+pub trait PseudoHeader {
+    /// Converts IPv4 or IPv6 specific fields to a checksum value
+    // Should we add a "next_header" u16 argument for ipv6 packets with extension header?
```

Comment:
the pseudo header is specific to ipv4 and ipv6... so having this be a trait with two implementations makes sense.